src/vmm/src/resources.rs

```diff
@@ -72,7 +73,7 @@ pub struct VmmConfig {
     #[serde(rename = "drives")]
     block_devices: Vec<BlockDeviceConfig>,
     #[serde(rename = "boot-source")]
-    boot_source: BootSourceConfig,
+    boot_source: Option<BootSourceConfig>,
```

Comment:
I'm having a hard time finding it now but there was some documentation edit suggestion in the past that used the word "empty" when referring to a couple of properties (one of which was `boot_source`). That made me think that in the case of restored snapshots, we'd want the `boot_source` to appear as null (since there isn't actually a boot source in that case). If we'd rather keep it as the default `BootSourceConfig` that's no problem, just let me know and I'll undo this change of making it `Optional`.